src/OrchardCore/OrchardCore.Recipes.Core/Services/RecipeExecutor.cs

```diff
@@ -157,11 +168,11 @@ public async Task<string> ExecuteAsync(string executionId, RecipeDescriptor reci
 
         private async Task ExecuteStepAsync(RecipeExecutionContext recipeStep)
         {
-            var (scope, shellContext) = await _orchardHost.GetScopeAndContextAsync(_shellSettings);
+            (var scope, var shellContext) = await _orchardHost.GetScopeAndContextAsync(_shellSettings);
 
             using (scope)
             {
-                if (!shellContext.IsActivated)
+                if (!recipeStep.IsMigration && !shellContext.IsActivated)
```

Comment:
Can't it be a metadata in the RecipeDesctriptor?

Reply:
OK that answer's that question ... I will add prop to RecipeDescriptor

Reply:
Don't name it with `Migration` in it, more something like `RequireNewScope`, so it can be used by something else than migrations.